crate_universe/extensions.bzl

```diff
@@ -658,8 +661,10 @@ def _generate_hub_and_spokes(
             "metadata": splice_outputs.metadata,
         })
 
-    # The workspace root when one is explicitly provided.
-    nonhermetic_root_bazel_workspace_dir = module_ctx.path(Label("@@//:MODULE.bazel")).dirname
+        for path_to_track in splice_outputs.extra_paths_to_track:
+            # We can only watch paths in our workspace.
```

Comment:
I believe this is what the old `module_ctx.read` stuff was for - it seems to successfully track external paths (or at least, not error on them, but IIRC I checked it seemed to work) whereas `module_ctx.watch` fails?

Reply:
see explanation in https://github.com/bazelbuild/rules_rust/pull/3569 - I think it never worked correctly (or the docs are wrong?). I bet @fmeum knows the answer?

Reply:
Read didn't error because it defaults to auto, but also won't watch.